File: Data Structures & Algorithms/is-palindrome/submission-3.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        var result = s.replace(/[^a-zA-Z0-9]/g, "").toLowerCase().trim()
        let left = 0;
        let right = result.length-1;
        while(left<right){
            if(result[left] !== result[right]){
                return false
            }
            left++;
            right--
        }
        return true
    }
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-3.java
class Solution {
    public boolean validPalindrome(String s) {
        int left = 0;
        int right = s.length()-1;

        while(left < right){
            if(s.charAt(left) != s.charAt(right)){
                return isValidPalindrome(s.substring(0,right) +s.substring(right+1)) ||
                        isValidPalindrome(s.substring(0,left) +s.substring(left+1));
            }
            left++;
            right--;
        }
        return true;
    }
    public boolean isValidPalindrome(String s){
        int left = 0;
        int right = s.length()-1;


        while(left < right){
            if(s.charAt(left) != s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}